--- a/commons/zenoh-util/src/net/mod.rs
+++ b/commons/zenoh-util/src/net/mod.rs
@@ -548,6 +548,22 @@ pub fn set_bind_to_device_tcp_socket(socket: &TcpSocket, iface: &str) -> ZResul
 pub fn set_bind_to_device_udp_socket(socket: &UdpSocket, iface: &str) -> ZResult<()> {
     tracing::warn!("Binding the socket {socket:?} to the interface {iface} is not supported on macOS, iOS, and Windows");
     Ok(())
 }
+
+#[cfg(target_os = "freebsd")]
+pub fn set_bind_to_device_tcp_socket(socket: &TcpSocket, iface: &str) -> ZResult<()> {
+    tracing::warn!("Binding the socket {socket:?} to the interface {iface} is not supported on FreeBSD");
+    Ok(())
+}
+
+#[cfg(target_os = "freebsd")]
+pub fn set_bind_to_device_udp_socket(socket: &UdpSocket, iface: &str) -> ZResult<()> {
+    tracing::warn!("Binding the socket {socket:?} to the interface {iface} is not supported on FreeBSD");
+    Ok(())
+}
